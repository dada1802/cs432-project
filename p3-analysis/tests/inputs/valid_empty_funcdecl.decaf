def int main()
{
    return 0;
}

def void ongo()
{

}