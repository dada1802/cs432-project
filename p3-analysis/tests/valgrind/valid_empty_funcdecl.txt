==137541== Memcheck, a memory error detector
==137541== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==137541== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==137541== Command: ../decaf inputs/valid_empty_funcdecl.decaf
==137541== 
Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 main : () -> int
 ongo : () -> void

  FuncDecl name="main" return_type=int parameters={} [line 1]
  SYM TABLE:

    Block [line 2]
    SYM TABLE:

  FuncDecl name="ongo" return_type=void parameters={} [line 6]
  SYM TABLE:

    Block [line 7]
    SYM TABLE:

==137541== 
==137541== HEAP SUMMARY:
==137541==     in use at exit: 0 bytes in 0 blocks
==137541==   total heap usage: 1,799 allocs, 1,799 frees, 213,824 bytes allocated
==137541== 
==137541== All heap blocks were freed -- no leaks are possible
==137541== 
==137541== For lists of detected and suppressed errors, rerun with: -s
==137541== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
