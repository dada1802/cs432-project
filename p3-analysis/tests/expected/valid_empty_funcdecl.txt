Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 main : () -> int
 ongo : () -> void

  FuncDecl name="main" return_type=int parameters={} [line 1]
  SYM TABLE:

    Block [line 2]
    SYM TABLE:

  FuncDecl name="ongo" return_type=void parameters={} [line 6]
  SYM TABLE:

    Block [line 7]
    SYM TABLE:

